### ISSUE TITLE
unittests for RST parsing should be based on the IR (intermediate AST)

### ISSUE THREAD

**Issue by timotheecour** [timotheecour/Nim#676]
Use an intermediate representation for RST unittests checks instead of going straight from input rst to output html, ie something like macros.treeRepr but for rst.

such checks could become easier to understand:

```
check input.toRstIR.toPretty == fmt"""
list("*")
  text("a")
  list("-m")
    text("desc")
  list("-n")
    text("very long desc")
list("-d")
  text("option")
"""
```

furthermore, they could be reused across backends (rst, latex), and wouldn't be affected by styling changes.

## note
* possibly `renderRstToStr` from rstast would be what we want here.
* we'd still have tests for html, but they'd be more of an integration test, and not written by hand but generated and visually checked when they need to change, as described here: https://github.com/nim-lang/Nim/pull/17442/files/c298a6d9a2e48763c65b390a057ae45b6075c504#r602769378

## links
* originally suggested here: https://github.com/nim-lang/Nim/pull/17442/files/c298a6d9a2e48763c65b390a057ae45b6075c504#r600893808

/cc @a-mr I really think we should do this, because tests like this are not very readable nor editable for humans:
```nim
  test "Field list":
    let output = ":field: text".toHtml
    check(output == """<table class="docinfo" frame="void" rules="none">""" &
            """<col class="docinfo-name" /><col class="docinfo-content" />""" &
            """<tbody valign="top"><tr><th class="docinfo-name">field:</th>""" &
            """<td> text</td></tr>""" & "\n</tbody></table>")

  test "Field list: body after newline":
    let output = dedent """
      :field:
        text1""".toHtml
    check "<table class=\"docinfo\"" in output
    check ">field:</th>" in output
    check "<td>text1</td>" in output
```
(as introduced in https://github.com/nim-lang/Nim/pull/17514)
(among other problems, you have to resort to splitting long lines here whereas this wouldn't be a problem with AST based tests)

**Comment by timotheecour** [comment]
@a-mr i like the tests involving `toAst` (eg https://github.com/nim-lang/Nim/pull/17908); maybe we should migrate to this kind of tests gradually (and still have 1 or a few html rendering tests that would be meant to be inspected visually when things must change) ?

This would cover what I had in mind with this issue.

**Comment by a-mr** [comment]
yes, trstgen.nim cries out to be refactored. And you phrased a good guiding principle on how to move the tests to trst.nim. But you know it's not of very high priority for me, may be in a few months...
